--- net/minecraft/world/level/levelgen/carver/CaveWorldCarver.java
+++ net/minecraft/world/level/levelgen/carver/CaveWorldCarver.java
@@ -26,12 +26,12 @@
 
       for(int var11 = 0; var11 < var10; ++var11) {
          double var12 = (double)var7.getBlockX(var5.nextInt(16));
          double var14 = (double)this.getCaveY(var1, var5);
          double var16 = (double)var7.getBlockZ(var5.nextInt(16));
-         double var18 = (double)Mth.randomBetween(var5, 0.2F, 1.8F);
-         double var20 = (double)Mth.randomBetween(var5, 0.2F, 1.8F);
+         double var18 = (double)Mth.randomBetween(var5, 0.3F, 1.7F);
+         double var20 = (double)Mth.randomBetween(var5, 0.3F, 1.7F);
          double var22 = (double)Mth.randomBetween(var5, -1.0F, 0.0F);
          WorldCarver.CarveSkipChecker var24 = (var2x, var3x, var5x, var7x, var9x) -> {
             return shouldSkip(var3x, var5x, var7x, var22);
          };
          int var25 = 1;
