--- com/mojang/realmsclient/util/task/GetServerDetailsTask.java
+++ com/mojang/realmsclient/util/task/GetServerDetailsTask.java
@@ -97,23 +97,23 @@
    }
 
    private RealmsLongConfirmationScreen resourcePackDownloadConfirmationScreen(RealmsServerAddress var1, Function<RealmsServerAddress, Screen> var2) {
       BooleanConsumer var3 = (var3x) -> {
          try {
-            if (var3x) {
-               this.scheduleResourcePackDownload(var1).thenRun(() -> {
-                  setScreen((Screen)var2.apply(var1));
-               }).exceptionally((var1x) -> {
-                  Minecraft.getInstance().getClientPackSource().clearServerPack();
-                  LOGGER.error(var1x);
-                  setScreen(new RealmsGenericErrorScreen(new TextComponent("Failed to download resource pack!"), this.lastScreen));
-                  return null;
-               });
+            if (!var3x) {
+               setScreen(this.lastScreen);
                return;
             }
 
-            setScreen(this.lastScreen);
+            this.scheduleResourcePackDownload(var1).thenRun(() -> {
+               setScreen((Screen)var2.apply(var1));
+            }).exceptionally((var1x) -> {
+               Minecraft.getInstance().getClientPackSource().clearServerPack();
+               LOGGER.error(var1x);
+               setScreen(new RealmsGenericErrorScreen(new TextComponent("Failed to download resource pack!"), this.lastScreen));
+               return null;
+            });
          } finally {
             if (this.connectLock.isHeldByCurrentThread()) {
                this.connectLock.unlock();
             }
 
