--- net/minecraft/client/renderer/debug/DebugRenderer.java
+++ net/minecraft/client/renderer/debug/DebugRenderer.java
@@ -167,33 +167,34 @@
       if (var13.isInitialized() && var12.getEntityRenderDispatcher().options != null) {
          Font var14 = var12.font;
          double var15 = var13.getPosition().x;
          double var17 = var13.getPosition().y;
          double var19 = var13.getPosition().z;
-         RenderSystem.pushMatrix();
-         RenderSystem.translatef((float)(var1 - var15), (float)(var3 - var17) + 0.07F, (float)(var5 - var19));
-         RenderSystem.normal3f(0.0F, 1.0F, 0.0F);
-         RenderSystem.multMatrix(new Matrix4f(var13.rotation()));
-         RenderSystem.scalef(var8, -var8, var8);
+         PoseStack var21 = RenderSystem.getModelViewStack();
+         var21.pushPose();
+         var21.translate((double)((float)(var1 - var15)), (double)((float)(var3 - var17) + 0.07F), (double)((float)(var5 - var19)));
+         var21.mulPoseMatrix(new Matrix4f(var13.rotation()));
+         var21.scale(var8, -var8, var8);
          RenderSystem.enableTexture();
          if (var11) {
             RenderSystem.disableDepthTest();
          } else {
             RenderSystem.enableDepthTest();
          }
 
          RenderSystem.depthMask(true);
-         RenderSystem.scalef(-1.0F, 1.0F, 1.0F);
-         float var21 = var9 ? (float)(-var14.width(var0)) / 2.0F : 0.0F;
-         var21 -= var10 / var8;
-         RenderSystem.enableAlphaTest();
-         MultiBufferSource.BufferSource var22 = MultiBufferSource.immediate(Tesselator.getInstance().getBuilder());
-         var14.drawInBatch((String)var0, var21, 0.0F, var7, false, Transformation.identity().getMatrix(), var22, var11, 0, 15728880);
-         var22.endBatch();
-         RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+         var21.scale(-1.0F, 1.0F, 1.0F);
+         RenderSystem.applyModelViewMatrix();
+         float var22 = var9 ? (float)(-var14.width(var0)) / 2.0F : 0.0F;
+         var22 -= var10 / var8;
+         MultiBufferSource.BufferSource var23 = MultiBufferSource.immediate(Tesselator.getInstance().getBuilder());
+         var14.drawInBatch((String)var0, var22, 0.0F, var7, false, Transformation.identity().getMatrix(), var23, var11, 0, 15728880);
+         var23.endBatch();
+         RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
          RenderSystem.enableDepthTest();
-         RenderSystem.popMatrix();
+         var21.popPose();
+         RenderSystem.applyModelViewMatrix();
       }
    }
 
    public interface SimpleDebugRenderer {
       void render(PoseStack var1, MultiBufferSource var2, double var3, double var5, double var7);
