--- net/minecraft/core/cauldron/CauldronInteraction.java
+++ net/minecraft/core/cauldron/CauldronInteraction.java
@@ -3,11 +3,10 @@
 import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
 import java.util.Map;
 import java.util.function.Predicate;
 import net.minecraft.Util;
 import net.minecraft.core.BlockPos;
-import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.sounds.SoundEvent;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.sounds.SoundSource;
 import net.minecraft.stats.Stats;
 import net.minecraft.world.InteractionHand;
@@ -76,11 +75,11 @@
             }
 
             if (var5.isEmpty()) {
                var3.setItemInHand(var4, var6);
             } else if (var3.getInventory().add(var6)) {
-               ((ServerPlayer)var3).refreshContainer(var3.inventoryMenu);
+               var3.inventoryMenu.sendAllDataToRemote();
             } else {
                var3.drop(var6, false);
             }
 
             var3.awardStat(Stats.CLEAN_BANNER);
