--- net/minecraft/data/models/ItemModelGenerators.java
+++ net/minecraft/data/models/ItemModelGenerators.java
@@ -242,10 +242,11 @@
       this.generateFlatItem(Items.SLIME_BALL, ModelTemplates.FLAT_ITEM);
       this.generateFlatItem(Items.SNOWBALL, ModelTemplates.FLAT_ITEM);
       this.generateFlatItem(Items.SPECTRAL_ARROW, ModelTemplates.FLAT_ITEM);
       this.generateFlatItem(Items.SPIDER_EYE, ModelTemplates.FLAT_ITEM);
       this.generateFlatItem(Items.SPRUCE_BOAT, ModelTemplates.FLAT_ITEM);
+      this.generateFlatItem(Items.SPYGLASS, ModelTemplates.FLAT_ITEM);
       this.generateFlatItem(Items.STICK, ModelTemplates.FLAT_HANDHELD_ITEM);
       this.generateFlatItem(Items.STONE_AXE, ModelTemplates.FLAT_HANDHELD_ITEM);
       this.generateFlatItem(Items.STONE_HOE, ModelTemplates.FLAT_HANDHELD_ITEM);
       this.generateFlatItem(Items.STONE_PICKAXE, ModelTemplates.FLAT_HANDHELD_ITEM);
       this.generateFlatItem(Items.STONE_SHOVEL, ModelTemplates.FLAT_HANDHELD_ITEM);
