--- net/minecraft/nbt/ListTag.java
+++ net/minecraft/nbt/ListTag.java
@@ -144,21 +144,10 @@
       }
 
       return 0;
    }
 
-   public int[] getIntArray(int var1) {
-      if (var1 >= 0 && var1 < this.list.size()) {
-         Tag var2 = (Tag)this.list.get(var1);
-         if (var2.getId() == 11) {
-            return ((IntArrayTag)var2).getAsIntArray();
-         }
-      }
-
-      return new int[0];
-   }
-
    public double getDouble(int var1) {
       if (var1 >= 0 && var1 < this.list.size()) {
          Tag var2 = (Tag)this.list.get(var1);
          if (var2.getId() == 6) {
             return ((DoubleTag)var2).getAsDouble();
