--- net/minecraft/world/inventory/CartographyTableMenu.java
+++ net/minecraft/world/inventory/CartographyTableMenu.java
@@ -51,11 +51,11 @@
       this.addSlot(new Slot(this.resultContainer, 2, 145, 39) {
          public boolean mayPlace(ItemStack var1) {
             return false;
          }
 
-         public ItemStack onTake(Player var1, ItemStack var2) {
+         public void onTake(Player var1, ItemStack var2) {
             ((Slot)CartographyTableMenu.this.slots.get(0)).remove(1);
             ((Slot)CartographyTableMenu.this.slots.get(1)).remove(1);
             var2.getItem().onCraftedBy(var2, var1.level, var1);
             var3.execute((var1x, var2x) -> {
                long var3x = var1x.getGameTime();
@@ -63,11 +63,11 @@
                   var1x.playSound((Player)null, (BlockPos)var2x, SoundEvents.UI_CARTOGRAPHY_TABLE_TAKE_RESULT, SoundSource.BLOCKS, 1.0F, 1.0F);
                   CartographyTableMenu.this.lastSoundTime = var3x;
                }
 
             });
-            return super.onTake(var1, var2);
+            super.onTake(var1, var2);
          }
       });
 
       int var4;
       for(var4 = 0; var4 < 3; ++var4) {
