--- net/minecraft/world/inventory/FurnaceResultSlot.java
+++ net/minecraft/world/inventory/FurnaceResultSlot.java
@@ -25,14 +25,13 @@
       }
 
       return super.remove(var1);
    }
 
-   public ItemStack onTake(Player var1, ItemStack var2) {
+   public void onTake(Player var1, ItemStack var2) {
       this.checkTakeAchievements(var2);
       super.onTake(var1, var2);
-      return var2;
    }
 
    protected void onQuickCraft(ItemStack var1, int var2) {
       this.removeCount += var2;
       this.checkTakeAchievements(var1);
