--- net/minecraft/world/inventory/ResultSlot.java
+++ net/minecraft/world/inventory/ResultSlot.java
@@ -48,11 +48,11 @@
       }
 
       this.removeCount = 0;
    }
 
-   public ItemStack onTake(Player var1, ItemStack var2) {
+   public void onTake(Player var1, ItemStack var2) {
       this.checkTakeAchievements(var2);
       NonNullList var3 = var1.level.getRecipeManager().getRemainingItemsFor(RecipeType.CRAFTING, this.craftSlots, var1.level);
 
       for(int var4 = 0; var4 < var3.size(); ++var4) {
          ItemStack var5 = this.craftSlots.getItem(var4);
@@ -72,8 +72,7 @@
                this.player.drop(var6, false);
             }
          }
       }
 
-      return var2;
    }
 }
