--- net/minecraft/world/level/block/grower/AbstractMegaTreeGrower.java
+++ net/minecraft/world/level/block/grower/AbstractMegaTreeGrower.java
@@ -34,11 +34,10 @@
    public boolean placeMega(ServerLevel var1, ChunkGenerator var2, BlockPos var3, BlockState var4, Random var5, int var6, int var7) {
       ConfiguredFeature var8 = this.getConfiguredMegaFeature(var5);
       if (var8 == null) {
          return false;
       } else {
-         ((TreeConfiguration)var8.config).setFromSapling();
          BlockState var9 = Blocks.AIR.defaultBlockState();
          var1.setBlock(var3.offset(var6, 0, var7), var9, 4);
          var1.setBlock(var3.offset(var6 + 1, 0, var7), var9, 4);
          var1.setBlock(var3.offset(var6, 0, var7 + 1), var9, 4);
          var1.setBlock(var3.offset(var6 + 1, 0, var7 + 1), var9, 4);
