--- net/minecraft/world/level/block/entity/SignBlockEntity.java
+++ net/minecraft/world/level/block/entity/SignBlockEntity.java
@@ -31,10 +31,11 @@
    private boolean isEditable;
    private Player playerWhoMayEdit;
    @Nullable
    private FormattedCharSequence[] renderMessages;
    private DyeColor color;
+   private boolean hasGlowingText;
 
    public SignBlockEntity(BlockPos var1, BlockState var2) {
       super(BlockEntityType.SIGN, var1, var2);
       this.messages = new Component[]{TextComponent.EMPTY, TextComponent.EMPTY, TextComponent.EMPTY, TextComponent.EMPTY};
       this.filteredMessages = new Component[]{TextComponent.EMPTY, TextComponent.EMPTY, TextComponent.EMPTY, TextComponent.EMPTY};
@@ -54,10 +55,11 @@
             var1.putString(FILTERED_TEXT_FIELD_NAMES[var2], Component.Serializer.toJson(var5));
          }
       }
 
       var1.putString("Color", this.color.getName());
+      var1.putBoolean("GlowingText", this.hasGlowingText);
       return var1;
    }
 
    public void load(CompoundTag var1) {
       this.isEditable = false;
@@ -75,10 +77,11 @@
             this.filteredMessages[var2] = var4;
          }
       }
 
       this.renderMessages = null;
+      this.hasGlowingText = var1.getBoolean("GlowingText");
    }
 
    private Component loadLine(String var1) {
       Component var2 = this.deserializeTextSafe(var1);
       if (this.level instanceof ServerLevel) {
@@ -169,13 +172,31 @@
    }
 
    public boolean setColor(DyeColor var1) {
       if (var1 != this.getColor()) {
          this.color = var1;
-         this.setChanged();
-         this.level.sendBlockUpdated(this.getBlockPos(), this.getBlockState(), this.getBlockState(), 3);
+         this.markUpdated();
          return true;
       } else {
          return false;
       }
    }
+
+   public boolean hasGlowingText() {
+      return this.hasGlowingText;
+   }
+
+   public boolean setHasGlowingText(boolean var1) {
+      if (this.hasGlowingText != var1) {
+         this.hasGlowingText = var1;
+         this.markUpdated();
+         return true;
+      } else {
+         return false;
+      }
+   }
+
+   private void markUpdated() {
+      this.setChanged();
+      this.level.sendBlockUpdated(this.getBlockPos(), this.getBlockState(), this.getBlockState(), 3);
+   }
 }
