--- net/minecraft/client/renderer/ItemInHandRenderer.java
+++ net/minecraft/client/renderer/ItemInHandRenderer.java
@@ -1,9 +1,10 @@
 package net.minecraft.client.renderer;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
+import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.blaze3d.vertex.VertexConsumer;
 import com.mojang.math.Matrix4f;
 import com.mojang.math.Vector3f;
 import net.minecraft.client.Minecraft;
@@ -58,11 +59,11 @@
       var2 = -Mth.cos(var2 * 3.1415927F) * 0.5F + 0.5F;
       return var2;
    }
 
    private void renderMapHand(PoseStack var1, MultiBufferSource var2, int var3, HumanoidArm var4) {
-      this.minecraft.getTextureManager().bind(this.minecraft.player.getSkinTextureLocation());
+      RenderSystem.setShaderTexture(0, this.minecraft.player.getSkinTextureLocation());
       PlayerRenderer var5 = (PlayerRenderer)this.entityRenderDispatcher.getRenderer(this.minecraft.player);
       var1.pushPose();
       float var6 = var4 == HumanoidArm.RIGHT ? 1.0F : -1.0F;
       var1.mulPose(Vector3f.YP.rotationDegrees(92.0F));
       var1.mulPose(Vector3f.XP.rotationDegrees(45.0F));
@@ -155,11 +156,11 @@
       float var13 = Mth.sin(var5 * var5 * 3.1415927F);
       float var14 = Mth.sin(var9 * 3.1415927F);
       var1.mulPose(Vector3f.YP.rotationDegrees(var8 * var14 * 70.0F));
       var1.mulPose(Vector3f.ZP.rotationDegrees(var8 * var13 * -20.0F));
       LocalPlayer var15 = this.minecraft.player;
-      this.minecraft.getTextureManager().bind(var15.getSkinTextureLocation());
+      RenderSystem.setShaderTexture(0, var15.getSkinTextureLocation());
       var1.translate((double)(var8 * -1.0F), 3.5999999046325684D, 3.5D);
       var1.mulPose(Vector3f.ZP.rotationDegrees(var8 * 120.0F));
       var1.mulPose(Vector3f.XP.rotationDegrees(200.0F));
       var1.mulPose(Vector3f.YP.rotationDegrees(var8 * -135.0F));
       var1.translate((double)(var8 * 5.6F), 0.0D, 0.0D);
