--- net/minecraft/data/worldgen/BiomeDefaultFeatures.java
+++ net/minecraft/data/worldgen/BiomeDefaultFeatures.java
@@ -151,10 +151,24 @@
 
    public static void addShatteredSavannaTrees(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TREES_SHATTERED_SAVANNA);
    }
 
+   public static void addLushCavesVegetationFeatures(BiomeGenerationSettings.Builder var0) {
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.LUSH_CAVES_CEILING_VEGETATION);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.CAVE_VINES);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.LUSH_CAVES_CLAY);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.LUSH_CAVES_VEGETATION);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.ROOTED_AZALEA_TREES);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.SPORE_BLOSSOM_FEATURE);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.CLASSIC_VINES_CAVE_FEATURE);
+   }
+
+   public static void addLushCavesSpecialOres(BiomeGenerationSettings.Builder var0) {
+      var0.addFeature(GenerationStep.Decoration.UNDERGROUND_ORES, Features.ORE_CLAY);
+   }
+
    public static void addMountainTrees(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TREES_MOUNTAIN);
    }
 
    public static void addMountainEdgeTrees(BiomeGenerationSettings.Builder var0) {
@@ -168,15 +182,15 @@
    public static void addJungleEdgeTrees(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TREES_JUNGLE_EDGE);
    }
 
    public static void addBadlandsTrees(BiomeGenerationSettings.Builder var0) {
-      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.OAK_BADLANDS);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TREES_BADLANDS);
    }
 
    public static void addSnowyTrees(BiomeGenerationSettings.Builder var0) {
-      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.SPRUCE_SNOWY);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TREES_SNOWY);
    }
 
    public static void addJungleGrass(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.PATCH_GRASS_JUNGLE);
    }
@@ -205,11 +219,11 @@
    public static void addForestGrass(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.PATCH_GRASS_FOREST);
    }
 
    public static void addSwampVegetation(BiomeGenerationSettings.Builder var0) {
-      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.SWAMP_TREE);
+      var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TREES_SWAMP);
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.FLOWER_SWAMP);
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.PATCH_GRASS_NORMAL);
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.PATCH_DEAD_BUSH);
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.PATCH_WATERLILLY);
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.BROWN_MUSHROOM_SWAMP);
@@ -296,11 +310,12 @@
    public static void addDesertExtraDecoration(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.SURFACE_STRUCTURES, Features.WELL);
    }
 
    public static void addFossilDecoration(BiomeGenerationSettings.Builder var0) {
-      var0.addFeature(GenerationStep.Decoration.UNDERGROUND_STRUCTURES, Features.FOSSIL);
+      var0.addFeature(GenerationStep.Decoration.UNDERGROUND_STRUCTURES, Features.FOSSIL_UPPER);
+      var0.addFeature(GenerationStep.Decoration.UNDERGROUND_STRUCTURES, Features.FOSSIL_LOWER);
    }
 
    public static void addColdOceanExtraVegetation(BiomeGenerationSettings.Builder var0) {
       var0.addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.KELP_COLD);
    }
