--- net/minecraft/network/protocol/game/ClientGamePacketListener.java
+++ net/minecraft/network/protocol/game/ClientGamePacketListener.java
@@ -37,12 +37,10 @@
 
    void handleChunkBlocksUpdate(ClientboundSectionBlocksUpdatePacket var1);
 
    void handleMapItemData(ClientboundMapItemDataPacket var1);
 
-   void handleContainerAck(ClientboundContainerAckPacket var1);
-
    void handleContainerClose(ClientboundContainerClosePacket var1);
 
    void handleContainerContent(ClientboundContainerSetContentPacket var1);
 
    void handleHorseScreenOpen(ClientboundHorseScreenOpenPacket var1);
