--- net/minecraft/world/item/BundleItem.java
+++ net/minecraft/world/item/BundleItem.java
@@ -13,12 +13,12 @@
 import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.stats.Stats;
 import net.minecraft.util.Mth;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.SlotAccess;
 import net.minecraft.world.entity.item.ItemEntity;
-import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.inventory.ClickAction;
 import net.minecraft.world.inventory.Slot;
 import net.minecraft.world.inventory.tooltip.BundleTooltip;
 import net.minecraft.world.inventory.tooltip.TooltipComponent;
@@ -33,32 +33,32 @@
 
    public static float getFullnessDisplay(ItemStack var0) {
       return (float)getContentWeight(var0) / 64.0F;
    }
 
-   public boolean overrideStackedOnOther(ItemStack var1, Slot var2, ClickAction var3, Inventory var4) {
+   public boolean overrideStackedOnOther(ItemStack var1, Slot var2, ClickAction var3, Player var4) {
       if (var3 != ClickAction.SECONDARY) {
          return false;
       } else {
          ItemStack var5 = var2.getItem();
          if (var5.isEmpty()) {
             removeOne(var1).ifPresent((var2x) -> {
                add(var1, var2.safeInsert(var2x));
             });
          } else if (var5.getItem().canFitInsideContainerItems()) {
             int var6 = (64 - getContentWeight(var1)) / getWeight(var5);
-            add(var1, var2.safeTake(var5.getCount(), var6, var4.player));
+            add(var1, var2.safeTake(var5.getCount(), var6, var4));
          }
 
          return true;
       }
    }
 
-   public boolean overrideOtherStackedOnMe(ItemStack var1, ItemStack var2, Slot var3, ClickAction var4, Inventory var5) {
-      if (var4 == ClickAction.SECONDARY && var3.allowModification(var5.player)) {
+   public boolean overrideOtherStackedOnMe(ItemStack var1, ItemStack var2, Slot var3, ClickAction var4, Player var5, SlotAccess var6) {
+      if (var4 == ClickAction.SECONDARY && var3.allowModification(var5)) {
          if (var2.isEmpty()) {
-            removeOne(var1).ifPresent(var5::setCarried);
+            removeOne(var1).ifPresent(var6::set);
          } else {
             var2.shrink(add(var1, var2));
          }
 
          return true;
