--- net/minecraft/world/level/levelgen/Aquifer.java
+++ net/minecraft/world/level/levelgen/Aquifer.java
@@ -113,11 +113,13 @@
       double var41 = this.similarity(var7, var8);
       double var42 = this.similarity(var7, var9);
       double var43 = this.similarity(var8, var9);
       this.lastWaterLevel = var16;
       this.shouldScheduleWaterUpdate = var41 > 0.0D;
-      if (var41 > -1.0D) {
+      if (this.lastWaterLevel >= var2 && var2 <= 9) {
+         this.lastBarrierDensity = 1.0D;
+      } else if (var41 > -1.0D) {
          double var44 = 1.0D + (this.barrierNoise.getValue((double)var1, (double)var2, (double)var3) + 0.1D) / 4.0D;
          double var46 = this.calculatePressure(var2, var44, var16, var17);
          double var47 = this.calculatePressure(var2, var44, var16, var18);
          double var31 = this.calculatePressure(var2, var44, var17, var18);
          double var33 = Math.max(0.0D, var41);
