--- net/minecraft/world/level/levelgen/NoiseSampler.java
+++ net/minecraft/world/level/levelgen/NoiseSampler.java
@@ -118,19 +118,19 @@
 
       for(int var36 = 0; var36 <= var7; ++var36) {
          int var37 = var36 + var6;
          double var38 = this.blendedNoise.sampleAndClampNoise(var2, var37, var3, var28, var29, var30, var32);
          double var39 = this.computeInitialDensity(var37, var8, var10, var34) + var38;
-         var39 = this.cavify(var2 * this.cellWidth, var37 * this.cellHeight, var3 * this.cellWidth, var38, var39);
+         var39 = this.cavify(var2 * this.cellWidth, var37 * this.cellHeight, var3 * this.cellWidth, var39);
          var39 = this.applySlide(var39, var37);
          var1[var36] = var39;
       }
 
    }
 
-   private double cavify(int var1, int var2, int var3, double var4, double var6) {
-      return this.cavifier != null ? this.cavifier.cavify(var1, var2, var3, var4, var6) : var6;
+   private double cavify(int var1, int var2, int var3, double var4) {
+      return this.cavifier != null ? this.cavifier.cavify(var1, var2, var3, var4) : var4;
    }
 
    private double computeInitialDensity(int var1, double var2, double var4, double var6) {
       double var8 = 1.0D - (double)var1 * 2.0D / 32.0D + var6;
       double var10 = var8 * this.dimensionDensityFactor + this.dimensionDensityOffset;
