--- net/minecraft/world/level/levelgen/SimpleRandomSource.java
+++ net/minecraft/world/level/levelgen/SimpleRandomSource.java
@@ -4,11 +4,10 @@
 import net.minecraft.util.DebugBuffer;
 import net.minecraft.util.ThreadingDetector;
 
 public class SimpleRandomSource implements RandomSource {
    private final AtomicLong seed = new AtomicLong();
-   private boolean haveNextNextGaussian = false;
 
    public SimpleRandomSource(long var1) {
       this.setSeed(var1);
    }
 
@@ -48,12 +47,18 @@
          return var3;
       }
    }
 
    public long nextLong() {
-      return ((long)this.nextInt() << 32) + (long)this.nextInt();
+      int var1 = this.next(32);
+      int var2 = this.next(32);
+      long var3 = (long)var1 << 32;
+      return var3 + (long)var2;
    }
 
    public double nextDouble() {
-      return (double)(((long)this.next(26) << 27) + (long)this.next(27)) * 1.1102230246251565E-16D;
+      int var1 = this.next(26);
+      int var2 = this.next(27);
+      long var3 = ((long)var1 << 27) + (long)var2;
+      return (double)var3 * 1.1102230246251565E-16D;
    }
 }
