--- net/minecraft/gametest/framework/TestCommand.java
+++ net/minecraft/gametest/framework/TestCommand.java
@@ -105,11 +105,11 @@
    private static int createNewStructure(CommandSourceStack var0, String var1, int var2, int var3, int var4) {
       if (var2 <= 48 && var3 <= 48 && var4 <= 48) {
          ServerLevel var5 = var0.getLevel();
          BlockPos var6 = new BlockPos(var0.getPosition());
          BlockPos var7 = new BlockPos(var6.getX(), var0.getLevel().getHeightmapPos(Heightmap.Types.WORLD_SURFACE, var6).getY(), var6.getZ() + 3);
-         StructureUtils.createNewEmptyStructureBlock(var1.toLowerCase(), var7, new BlockPos(var2, var3, var4), Rotation.NONE, var5);
+         StructureUtils.createNewEmptyStructureBlock(var1.toLowerCase(), var7, new Vec3i(var2, var3, var4), Rotation.NONE, var5);
 
          for(int var8 = 0; var8 < var2; ++var8) {
             for(int var9 = 0; var9 < var4; ++var9) {
                BlockPos var10 = new BlockPos(var7.getX() + var8, var7.getY() + 1, var7.getZ() + var9);
                Block var11 = Blocks.POLISHED_ANDESITE;
