--- net/minecraft/data/tags/BlockTagsProvider.java
+++ net/minecraft/data/tags/BlockTagsProvider.java
@@ -87,10 +87,13 @@
       this.tag(BlockTags.HOGLIN_REPELLENTS).add((Object)Blocks.WARPED_FUNGUS).add((Object)Blocks.POTTED_WARPED_FUNGUS).add((Object)Blocks.NETHER_PORTAL).add((Object)Blocks.RESPAWN_ANCHOR);
       this.tag(BlockTags.GOLD_ORES).add((Object[])(Blocks.GOLD_ORE, Blocks.NETHER_GOLD_ORE, Blocks.DEEPSLATE_GOLD_ORE));
       this.tag(BlockTags.IRON_ORES).add((Object[])(Blocks.IRON_ORE, Blocks.DEEPSLATE_IRON_ORE));
       this.tag(BlockTags.DIAMOND_ORES).add((Object[])(Blocks.DIAMOND_ORE, Blocks.DEEPSLATE_DIAMOND_ORE));
       this.tag(BlockTags.REDSTONE_ORES).add((Object[])(Blocks.REDSTONE_ORE, Blocks.DEEPSLATE_REDSTONE_ORE));
+      this.tag(BlockTags.COAL_ORES).add((Object[])(Blocks.COAL_ORE, Blocks.DEEPSLATE_COAL_ORE));
+      this.tag(BlockTags.EMERALD_ORES).add((Object[])(Blocks.EMERALD_ORE, Blocks.DEEPSLATE_EMERALD_ORE));
+      this.tag(BlockTags.COPPER_ORES).add((Object[])(Blocks.COPPER_ORE, Blocks.DEEPSLATE_COPPER_ORE));
       this.tag(BlockTags.LAPIS_ORES).add((Object[])(Blocks.LAPIS_ORE, Blocks.DEEPSLATE_LAPIS_ORE));
       this.tag(BlockTags.SOUL_FIRE_BASE_BLOCKS).add((Object[])(Blocks.SOUL_SAND, Blocks.SOUL_SOIL));
       this.tag(BlockTags.NON_FLAMMABLE_WOOD).add((Object[])(Blocks.WARPED_STEM, Blocks.STRIPPED_WARPED_STEM, Blocks.WARPED_HYPHAE, Blocks.STRIPPED_WARPED_HYPHAE, Blocks.CRIMSON_STEM, Blocks.STRIPPED_CRIMSON_STEM, Blocks.CRIMSON_HYPHAE, Blocks.STRIPPED_CRIMSON_HYPHAE, Blocks.CRIMSON_PLANKS, Blocks.WARPED_PLANKS, Blocks.CRIMSON_SLAB, Blocks.WARPED_SLAB, Blocks.CRIMSON_PRESSURE_PLATE, Blocks.WARPED_PRESSURE_PLATE, Blocks.CRIMSON_FENCE, Blocks.WARPED_FENCE, Blocks.CRIMSON_TRAPDOOR, Blocks.WARPED_TRAPDOOR, Blocks.CRIMSON_FENCE_GATE, Blocks.WARPED_FENCE_GATE, Blocks.CRIMSON_STAIRS, Blocks.WARPED_STAIRS, Blocks.CRIMSON_BUTTON, Blocks.WARPED_BUTTON, Blocks.CRIMSON_DOOR, Blocks.WARPED_DOOR, Blocks.CRIMSON_SIGN, Blocks.WARPED_SIGN, Blocks.CRIMSON_WALL_SIGN, Blocks.WARPED_WALL_SIGN));
       this.tag(BlockTags.STRIDER_WARM_BLOCKS).add((Object)Blocks.LAVA);
       this.tag(BlockTags.CAMPFIRES).add((Object[])(Blocks.CAMPFIRE, Blocks.SOUL_CAMPFIRE));
@@ -111,12 +114,12 @@
       this.tag(BlockTags.CRYSTAL_SOUND_BLOCKS).add((Object[])(Blocks.AMETHYST_BLOCK, Blocks.BUDDING_AMETHYST));
       this.tag(BlockTags.CAULDRONS).add((Object[])(Blocks.CAULDRON, Blocks.WATER_CAULDRON, Blocks.LAVA_CAULDRON, Blocks.POWDER_SNOW_CAULDRON));
       this.tag(BlockTags.INSIDE_STEP_SOUND_BLOCKS).add((Object[])(Blocks.SNOW, Blocks.POWDER_SNOW));
       this.tag(BlockTags.DRIPSTONE_REPLACEABLE).addTag(BlockTags.BASE_STONE_OVERWORLD).add((Object)Blocks.DIRT);
       this.tag(BlockTags.CAVE_VINES).add((Object)Blocks.CAVE_VINES_BODY).add((Object)Blocks.CAVE_VINES_HEAD);
-      this.tag(BlockTags.LUSH_PLANTS_REPLACEABLE).addTag(BlockTags.BASE_STONE_OVERWORLD).addTag(BlockTags.CAVE_VINES).add((Object)Blocks.DIRT).add((Object)Blocks.GRAVEL).add((Object)Blocks.SAND).add((Object)Blocks.MOSS_BLOCK).addTag(BlockTags.FLOWERS).add((Object)Blocks.GRASS).add((Object)Blocks.TALL_GRASS).add((Object)Blocks.MOSS_CARPET).add((Object)Blocks.SMALL_DRIPLEAF).add((Object)Blocks.BIG_DRIPLEAF).add((Object)Blocks.BIG_DRIPLEAF_STEM).add((Object)Blocks.VINE);
-      this.tag(BlockTags.AZALEA_LOG_REPLACEABLE).addTag(BlockTags.FLOWERS).addTag(BlockTags.LEAVES).add((Object)Blocks.GRASS).add((Object)Blocks.FERN).add((Object)Blocks.SWEET_BERRY_BUSH).add((Object)Blocks.SMALL_DRIPLEAF);
+      this.tag(BlockTags.LUSH_PLANTS_REPLACEABLE).addTag(BlockTags.BASE_STONE_OVERWORLD).addTag(BlockTags.CAVE_VINES).add((Object)Blocks.DIRT).add((Object)Blocks.GRAVEL).add((Object)Blocks.SAND);
+      this.tag(BlockTags.LUSH_GROUND_REPLACEABLE).addTag(BlockTags.LUSH_PLANTS_REPLACEABLE).add((Object)Blocks.CLAY).add((Object)Blocks.MOSS_BLOCK);
       this.tag(BlockTags.OCCLUDES_VIBRATION_SIGNALS).addTag(BlockTags.WOOL);
    }
 
    protected Path getPath(ResourceLocation var1) {
       return this.generator.getOutputFolder().resolve("data/" + var1.getNamespace() + "/tags/blocks/" + var1.getPath() + ".json");
