--- net/minecraft/data/tags/ItemTagsProvider.java
+++ net/minecraft/data/tags/ItemTagsProvider.java
@@ -61,10 +61,13 @@
       this.copy(BlockTags.GOLD_ORES, ItemTags.GOLD_ORES);
       this.copy(BlockTags.IRON_ORES, ItemTags.IRON_ORES);
       this.copy(BlockTags.DIAMOND_ORES, ItemTags.DIAMOND_ORES);
       this.copy(BlockTags.REDSTONE_ORES, ItemTags.REDSTONE_ORES);
       this.copy(BlockTags.LAPIS_ORES, ItemTags.LAPIS_ORES);
+      this.copy(BlockTags.COAL_ORES, ItemTags.COAL_ORES);
+      this.copy(BlockTags.EMERALD_ORES, ItemTags.EMERALD_ORES);
+      this.copy(BlockTags.COPPER_ORES, ItemTags.COPPER_ORES);
       this.copy(BlockTags.SOUL_FIRE_BASE_BLOCKS, ItemTags.SOUL_FIRE_BASE_BLOCKS);
       this.copy(BlockTags.CANDLES, ItemTags.CANDLES);
       this.copy(BlockTags.OCCLUDES_VIBRATION_SIGNALS, ItemTags.OCCLUDES_VIBRATION_SIGNALS);
       this.tag(ItemTags.BANNERS).add((Object[])(Items.WHITE_BANNER, Items.ORANGE_BANNER, Items.MAGENTA_BANNER, Items.LIGHT_BLUE_BANNER, Items.YELLOW_BANNER, Items.LIME_BANNER, Items.PINK_BANNER, Items.GRAY_BANNER, Items.LIGHT_GRAY_BANNER, Items.CYAN_BANNER, Items.PURPLE_BANNER, Items.BLUE_BANNER, Items.BROWN_BANNER, Items.GREEN_BANNER, Items.RED_BANNER, Items.BLACK_BANNER));
       this.tag(ItemTags.BOATS).add((Object[])(Items.OAK_BOAT, Items.SPRUCE_BOAT, Items.BIRCH_BOAT, Items.JUNGLE_BOAT, Items.ACACIA_BOAT, Items.DARK_OAK_BOAT));
