--- net/minecraft/tags/ItemTags.java
+++ net/minecraft/tags/ItemTags.java
@@ -51,10 +51,13 @@
    public static final Tag.Named<Item> GOLD_ORES;
    public static final Tag.Named<Item> IRON_ORES;
    public static final Tag.Named<Item> DIAMOND_ORES;
    public static final Tag.Named<Item> REDSTONE_ORES;
    public static final Tag.Named<Item> LAPIS_ORES;
+   public static final Tag.Named<Item> COAL_ORES;
+   public static final Tag.Named<Item> EMERALD_ORES;
+   public static final Tag.Named<Item> COPPER_ORES;
    public static final Tag.Named<Item> NON_FLAMMABLE_WOOD;
    public static final Tag.Named<Item> SOUL_FIRE_BASE_BLOCKS;
    public static final Tag.Named<Item> CANDLES;
    public static final Tag.Named<Item> BOATS;
    public static final Tag.Named<Item> FISHES;
@@ -127,10 +130,13 @@
       GOLD_ORES = bind("gold_ores");
       IRON_ORES = bind("iron_ores");
       DIAMOND_ORES = bind("diamond_ores");
       REDSTONE_ORES = bind("redstone_ores");
       LAPIS_ORES = bind("lapis_ores");
+      COAL_ORES = bind("coal_ores");
+      EMERALD_ORES = bind("emerald_ores");
+      COPPER_ORES = bind("copper_ores");
       NON_FLAMMABLE_WOOD = bind("non_flammable_wood");
       SOUL_FIRE_BASE_BLOCKS = bind("soul_fire_base_blocks");
       CANDLES = bind("candles");
       BOATS = bind("boats");
       FISHES = bind("fishes");
