--- net/minecraft/world/level/levelgen/structure/BuriedTreasurePieces.java
+++ net/minecraft/world/level/levelgen/structure/BuriedTreasurePieces.java
@@ -17,11 +17,11 @@
 
 public class BuriedTreasurePieces {
    public static class BuriedTreasurePiece extends StructurePiece {
       public BuriedTreasurePiece(BlockPos var1) {
          super(StructurePieceType.BURIED_TREASURE_PIECE, 0);
-         this.boundingBox = new BoundingBox(var1.getX(), var1.getY(), var1.getZ(), var1.getX(), var1.getY(), var1.getZ());
+         this.boundingBox = new BoundingBox(var1);
       }
 
       public BuriedTreasurePiece(StructureManager var1, CompoundTag var2) {
          super(StructurePieceType.BURIED_TREASURE_PIECE, var2);
       }
@@ -54,11 +54,11 @@
                         var1.setBlock(var17, var12, 3);
                      }
                   }
                }
 
-               this.boundingBox = new BoundingBox(var9.getX(), var9.getY(), var9.getZ(), var9.getX(), var9.getY(), var9.getZ());
+               this.boundingBox = new BoundingBox(var9);
                return this.createChest(var1, var5, var4, var9, BuiltInLootTables.BURIED_TREASURE, (BlockState)null);
             }
 
             var9.move(0, -1, 0);
          }
