--- net/minecraft/data/worldgen/biome/VanillaBiomes.java
+++ net/minecraft/data/worldgen/biome/VanillaBiomes.java
@@ -881,10 +881,29 @@
       var1.addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.SPRING_OPEN).addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.PATCH_FIRE).addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.PATCH_SOUL_FIRE).addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.GLOWSTONE_EXTRA).addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.GLOWSTONE).addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.ORE_MAGMA).addFeature(GenerationStep.Decoration.UNDERGROUND_DECORATION, Features.SPRING_CLOSED).addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.WARPED_FUNGI).addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.WARPED_FOREST_VEGETATION).addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.NETHER_SPROUTS).addFeature(GenerationStep.Decoration.VEGETAL_DECORATION, Features.TWISTING_VINES);
       BiomeDefaultFeatures.addNetherDefaultOres(var1);
       return (new Biome.BiomeBuilder()).precipitation(Biome.Precipitation.NONE).biomeCategory(Biome.BiomeCategory.NETHER).depth(0.1F).scale(0.2F).temperature(2.0F).downfall(0.0F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4159204).waterFogColor(329011).fogColor(1705242).skyColor(calculateSkyColor(2.0F)).ambientParticle(new AmbientParticleSettings(ParticleTypes.WARPED_SPORE, 0.01428F)).ambientLoopSound(SoundEvents.AMBIENT_WARPED_FOREST_LOOP).ambientMoodSound(new AmbientMoodSettings(SoundEvents.AMBIENT_WARPED_FOREST_MOOD, 6000, 8, 2.0D)).ambientAdditionsSound(new AmbientAdditionsSettings(SoundEvents.AMBIENT_WARPED_FOREST_ADDITIONS, 0.0111D)).backgroundMusic(Musics.createGameMusic(SoundEvents.MUSIC_BIOME_WARPED_FOREST)).build()).mobSpawnSettings(var0).generationSettings(var1.build()).build();
    }
 
+   public static Biome lushCaves() {
+      MobSpawnSettings.Builder var0 = new MobSpawnSettings.Builder();
+      BiomeDefaultFeatures.commonSpawns(var0);
+      BiomeGenerationSettings.Builder var1 = (new BiomeGenerationSettings.Builder()).surfaceBuilder(SurfaceBuilders.GRASS);
+      BiomeDefaultFeatures.addDefaultOverworldLandStructures(var1);
+      var1.addStructureStart(StructureFeatures.RUINED_PORTAL_STANDARD);
+      BiomeDefaultFeatures.addDefaultCarvers(var1);
+      BiomeDefaultFeatures.addDefaultLakes(var1);
+      BiomeDefaultFeatures.addDefaultCrystalFormations(var1);
+      BiomeDefaultFeatures.addDefaultMonsterRoom(var1);
+      BiomeDefaultFeatures.addPlainGrass(var1);
+      BiomeDefaultFeatures.addDefaultUndergroundVariety(var1);
+      BiomeDefaultFeatures.addDefaultOres(var1);
+      BiomeDefaultFeatures.addLushCavesSpecialOres(var1);
+      BiomeDefaultFeatures.addDefaultSoftDisks(var1);
+      BiomeDefaultFeatures.addLushCavesVegetationFeatures(var1);
+      return (new Biome.BiomeBuilder()).precipitation(Biome.Precipitation.RAIN).biomeCategory(Biome.BiomeCategory.UNDERGROUND).depth(0.1F).scale(0.2F).temperature(0.5F).downfall(0.5F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4159204).waterFogColor(329011).fogColor(12638463).skyColor(calculateSkyColor(0.5F)).build()).mobSpawnSettings(var0.build()).generationSettings(var1.build()).build();
+   }
+
    public static Biome dripstoneCaves() {
       MobSpawnSettings.Builder var0 = new MobSpawnSettings.Builder();
       BiomeDefaultFeatures.commonSpawns(var0);
       BiomeGenerationSettings.Builder var1 = (new BiomeGenerationSettings.Builder()).surfaceBuilder(SurfaceBuilders.GRASS);
       BiomeDefaultFeatures.addDefaultOverworldLandStructures(var1);
@@ -901,8 +920,8 @@
       BiomeDefaultFeatures.addDefaultMushrooms(var1);
       BiomeDefaultFeatures.addDefaultExtraVegetation(var1);
       BiomeDefaultFeatures.addDefaultSprings(var1);
       BiomeDefaultFeatures.addSurfaceFreezing(var1);
       BiomeDefaultFeatures.addDripstone(var1);
-      return (new Biome.BiomeBuilder()).precipitation(Biome.Precipitation.RAIN).biomeCategory(Biome.BiomeCategory.PLAINS).depth(0.125F).scale(0.05F).temperature(0.8F).downfall(0.4F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4159204).waterFogColor(329011).fogColor(12638463).skyColor(calculateSkyColor(0.8F)).ambientMoodSound(AmbientMoodSettings.LEGACY_CAVE_SETTINGS).build()).mobSpawnSettings(var0.build()).generationSettings(var1.build()).build();
+      return (new Biome.BiomeBuilder()).precipitation(Biome.Precipitation.RAIN).biomeCategory(Biome.BiomeCategory.UNDERGROUND).depth(0.125F).scale(0.05F).temperature(0.8F).downfall(0.4F).specialEffects((new BiomeSpecialEffects.Builder()).waterColor(4159204).waterFogColor(329011).fogColor(12638463).skyColor(calculateSkyColor(0.8F)).ambientMoodSound(AmbientMoodSettings.LEGACY_CAVE_SETTINGS).build()).mobSpawnSettings(var0.build()).generationSettings(var1.build()).build();
    }
 }
