--- net/minecraft/world/inventory/InventoryMenu.java
+++ net/minecraft/world/inventory/InventoryMenu.java
@@ -91,11 +91,11 @@
    public boolean recipeMatches(Recipe<? super CraftingContainer> var1) {
       return var1.matches(this.craftSlots, this.owner.level);
    }
 
    public void slotsChanged(Container var1) {
-      CraftingMenu.slotChangedCraftingGrid(this.containerId, this.owner.level, this.owner, this.craftSlots, this.resultSlots);
+      CraftingMenu.slotChangedCraftingGrid(this, this.owner.level, this.owner, this.craftSlots, this.resultSlots);
    }
 
    public void removed(Player var1) {
       super.removed(var1);
       this.resultSlots.clearContent();
@@ -158,13 +158,13 @@
 
          if (var5.getCount() == var3.getCount()) {
             return ItemStack.EMPTY;
          }
 
-         ItemStack var8 = var4.onTake(var1, var5);
+         var4.onTake(var1, var5);
          if (var2 == 0) {
-            var1.drop(var8, false);
+            var1.drop(var5, false);
          }
       }
 
       return var3;
    }
