--- net/minecraft/DetectedVersion.java
+++ net/minecraft/DetectedVersion.java
@@ -28,13 +28,13 @@
    private final Date buildTime;
    private final String releaseTarget;
 
    private DetectedVersion() {
       this.id = UUID.randomUUID().toString().replaceAll("-", "");
-      this.name = "21w08b";
+      this.name = "21w10a";
       this.stable = false;
-      this.worldVersion = 2698;
+      this.worldVersion = 2699;
       this.protocolVersion = SharedConstants.getProtocolVersion();
       this.resourcePackVersion = 7;
       this.dataPackVersion = 7;
       this.buildTime = new Date();
       this.releaseTarget = "1.17";
