--- net/minecraft/client/gui/screens/Screen.java
+++ net/minecraft/client/gui/screens/Screen.java
@@ -33,10 +33,11 @@
 import net.minecraft.client.gui.components.TickableWidget;
 import net.minecraft.client.gui.components.Widget;
 import net.minecraft.client.gui.components.events.AbstractContainerEventHandler;
 import net.minecraft.client.gui.components.events.GuiEventListener;
 import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
+import net.minecraft.client.renderer.GameRenderer;
 import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.entity.ItemRenderer;
 import net.minecraft.network.chat.ClickEvent;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.HoverEvent;
@@ -175,10 +176,11 @@
          boolean var14 = true;
          float var15 = this.itemRenderer.blitOffset;
          this.itemRenderer.blitOffset = 400.0F;
          Tesselator var16 = Tesselator.getInstance();
          BufferBuilder var17 = var16.getBuilder();
+         RenderSystem.setShader(GameRenderer::getPositionColorShader);
          var17.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_COLOR);
          Matrix4f var18 = var1.last().pose();
          fillGradient(var18, var17, var23 - 3, var24 - 4, var23 + var5 + 3, var24 - 3, 400, -267386864, -267386864);
          fillGradient(var18, var17, var23 - 3, var24 + var6 + 3, var23 + var5 + 3, var24 + var6 + 4, 400, -267386864, -267386864);
          fillGradient(var18, var17, var23 - 3, var24 - 3, var23 + var5 + 3, var24 + var6 + 3, 400, -267386864, -267386864);
@@ -190,14 +192,12 @@
          fillGradient(var18, var17, var23 - 3, var24 + var6 + 2, var23 + var5 + 3, var24 + var6 + 3, 400, 1344798847, 1344798847);
          RenderSystem.enableDepthTest();
          RenderSystem.disableTexture();
          RenderSystem.enableBlend();
          RenderSystem.defaultBlendFunc();
-         RenderSystem.shadeModel(7425);
          var17.end();
          BufferUploader.end(var17);
-         RenderSystem.shadeModel(7424);
          RenderSystem.disableBlend();
          RenderSystem.enableTexture();
          MultiBufferSource.BufferSource var19 = MultiBufferSource.immediate(Tesselator.getInstance().getBuilder());
          var1.translate(0.0D, 0.0D, 400.0D);
          int var20 = var24;
@@ -357,12 +357,13 @@
    }
 
    public void renderDirtBackground(int var1) {
       Tesselator var2 = Tesselator.getInstance();
       BufferBuilder var3 = var2.getBuilder();
-      this.minecraft.getTextureManager().bind(BACKGROUND_LOCATION);
-      RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+      RenderSystem.setShader(GameRenderer::getPositionTexColorShader);
+      RenderSystem.setShaderTexture(0, BACKGROUND_LOCATION);
+      RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
       float var4 = 32.0F;
       var3.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_TEX_COLOR);
       var3.vertex(0.0D, (double)this.height, 0.0D).uv(0.0F, (float)this.height / 32.0F + (float)var1).color(64, 64, 64, 255).endVertex();
       var3.vertex((double)this.width, (double)this.height, 0.0D).uv((float)this.width / 32.0F, (float)this.height / 32.0F + (float)var1).color(64, 64, 64, 255).endVertex();
       var3.vertex((double)this.width, 0.0D, 0.0D).uv((float)this.width / 32.0F, (float)var1).color(64, 64, 64, 255).endVertex();
