--- com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -353,14 +353,14 @@
          this.drawImage(var1, var3, var4 + 1, var5, var6, var2);
       }
 
       private void drawImage(PoseStack var1, int var2, int var3, int var4, int var5, WorldTemplate var6) {
          RealmsTextureManager.bindWorldTemplate(var6.id, var6.image);
-         RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+         RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
          GuiComponent.blit(var1, var2 + 1, var3 + 1, 0.0F, 0.0F, 38, 38, 38, 38);
-         RealmsSelectWorldTemplateScreen.this.minecraft.getTextureManager().bind(RealmsSelectWorldTemplateScreen.SLOT_FRAME_LOCATION);
-         RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+         RenderSystem.setShaderTexture(0, RealmsSelectWorldTemplateScreen.SLOT_FRAME_LOCATION);
+         RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
          GuiComponent.blit(var1, var2, var3, 0.0F, 0.0F, 40, 40, 40, 40);
       }
 
       private void drawIcons(PoseStack var1, int var2, int var3, int var4, int var5, String var6, String var7, String var8) {
          if (!"".equals(var8)) {
@@ -382,28 +382,22 @@
                var11 = true;
             }
          }
 
          if (!var12) {
-            RealmsSelectWorldTemplateScreen.this.minecraft.getTextureManager().bind(RealmsSelectWorldTemplateScreen.LINK_ICON);
-            RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
-            RenderSystem.pushMatrix();
-            RenderSystem.scalef(1.0F, 1.0F, 1.0F);
+            RenderSystem.setShaderTexture(0, RealmsSelectWorldTemplateScreen.LINK_ICON);
+            RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
             float var13 = var10 ? 15.0F : 0.0F;
             GuiComponent.blit(var1, var2 + var9, var3, var13, 0.0F, 15, 15, 30, 15);
-            RenderSystem.popMatrix();
          }
 
          if (!"".equals(var7)) {
-            RealmsSelectWorldTemplateScreen.this.minecraft.getTextureManager().bind(RealmsSelectWorldTemplateScreen.TRAILER_ICON);
-            RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
-            RenderSystem.pushMatrix();
-            RenderSystem.scalef(1.0F, 1.0F, 1.0F);
+            RenderSystem.setShaderTexture(0, RealmsSelectWorldTemplateScreen.TRAILER_ICON);
+            RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
             int var15 = var2 + var9 + (var12 ? 0 : 17);
             float var14 = var11 ? 15.0F : 0.0F;
             GuiComponent.blit(var1, var15, var3, var14, 0.0F, 15, 15, 30, 15);
-            RenderSystem.popMatrix();
          }
 
          if (var10) {
             RealmsSelectWorldTemplateScreen.this.toolTip = RealmsSelectWorldTemplateScreen.PUBLISHER_LINK_TOOLTIP;
             RealmsSelectWorldTemplateScreen.this.currentLink = var6;
