--- com/mojang/realmsclient/gui/screens/RealmsResetWorldScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsResetWorldScreen.java
@@ -160,23 +160,23 @@
       this.subtitleLabel.render(this, var1);
       super.render(var1, var2, var3, var4);
    }
 
    private void drawFrame(PoseStack var1, int var2, int var3, Component var4, ResourceLocation var5, boolean var6, boolean var7) {
-      this.minecraft.getTextureManager().bind(var5);
+      RenderSystem.setShaderTexture(0, var5);
       if (var6) {
-         RenderSystem.color4f(0.56F, 0.56F, 0.56F, 1.0F);
+         RenderSystem.setShaderColor(0.56F, 0.56F, 0.56F, 1.0F);
       } else {
-         RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+         RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
       }
 
       GuiComponent.blit(var1, var2 + 2, var3 + 14, 0.0F, 0.0F, 56, 56, 56, 56);
-      this.minecraft.getTextureManager().bind(SLOT_FRAME_LOCATION);
+      RenderSystem.setShaderTexture(0, SLOT_FRAME_LOCATION);
       if (var6) {
-         RenderSystem.color4f(0.56F, 0.56F, 0.56F, 1.0F);
+         RenderSystem.setShaderColor(0.56F, 0.56F, 0.56F, 1.0F);
       } else {
-         RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+         RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
       }
 
       GuiComponent.blit(var1, var2, var3 + 12, 0.0F, 0.0F, 60, 60, 60, 60);
       int var8 = var6 ? 10526880 : 16777215;
       drawCenteredString(var1, this.font, var4, var2 + 30, var3, var8);
