--- net/minecraft/client/renderer/blockentity/SignRenderer.java
+++ net/minecraft/client/renderer/blockentity/SignRenderer.java
@@ -88,11 +88,11 @@
       });
 
       for(int var23 = 0; var23 < 4; ++var23) {
          FormattedCharSequence var24 = var22[var23];
          float var25 = (float)(-this.font.width(var24) / 2);
-         int var26 = (Boolean)var7.getValue(SignBlock.LIT) ? 15728880 : var5;
+         int var26 = var1.hasGlowingText() ? 15728880 : var5;
          this.font.drawInBatch((FormattedCharSequence)var24, var25, (float)(var23 * 10 - 20), var20, false, var3.last().pose(), var4, false, 0, var26);
       }
 
       var3.popPose();
    }
