--- net/minecraft/server/commands/LootCommand.java
+++ net/minecraft/server/commands/LootCommand.java
@@ -219,13 +219,12 @@
 
       while(var6.hasNext()) {
          Entity var7 = (Entity)var6.next();
          if (var7 instanceof ServerPlayer) {
             ServerPlayer var8 = (ServerPlayer)var7;
-            var8.inventoryMenu.broadcastChanges();
             setSlots(var7, var3, var1, var2, var5);
-            var8.inventoryMenu.broadcastChanges();
+            var8.containerMenu.broadcastChanges();
          } else {
             setSlots(var7, var3, var1, var2, var5);
          }
       }
 
